ark/analysis/spatial_analysis.py

```diff
@@ -593,17 +602,22 @@ def generate_cluster_matrix_results(all_data, neighbor_mat, cluster_num, exclude
         raise ValueError("Invalid k provided for clustering")
 
     # subset neighbor mat
-    neighbor_mat_data = neighbor_mat[neighbor_mat[fov_col].isin(included_fovs)]
-    neighbor_mat_data = neighbor_mat_data.drop(fov_col, axis=1)
+    neighbor_mat_data_all = neighbor_mat[neighbor_mat[fov_col].isin(included_fovs)]
+    neighbor_mat_data = neighbor_mat_data_all.drop([fov_col, label_col], axis=1)
 
     # generate cluster labels
     cluster_labels = spatial_analysis_utils.generate_cluster_labels(
         neighbor_mat_data, cluster_num)
 
-    all_data_clusters = all_data.copy()
+    # add labels to neighbor mat, add 1 to avoid cluster number 0
+    neighbor_mat_data_all[cluster_label_col] = [x+1 for x in cluster_labels]
```

Comment:
Will you open an issue saying that `generate_cluster_labels` should default to 1-based cluster indices? Don't need to change it here if you don't want, but it would be a good default behavior to have eventually

Reply:
I'll just change it.